src/store/tickets/thunk.ts

```diff
@@ -0,0 +1,66 @@
+import { Notify } from '../../utils/notify';
+import TicketService from '../../service/TicketService';
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
+import { ITickets, IUpdateTickets } from '../../types/tickets';
+
+export const fetchTicketAsync = createAsyncThunk('tickets/fetchTicket', async (apiUrl: string) => {
+  try {
+    const response = await TicketService.getAll(apiUrl);
+    const data = response.data;
+    const totalCount = parseInt(response.headers['x-total-count']);
+
+    return { data, totalCount };
+  } catch (error) {
+    Notify('Ticket creation error');
+  }
+});
+export const createTicketAsync = createAsyncThunk(
+  'tickets/createTicket',
+  async ({ apiUrl, data }: { apiUrl: string; data: ITickets }, thunkAPI) => {
+    const { dispatch } = thunkAPI;
+    const transformedData = {
+      ...data,
+      createdBy: {
+        name: 'userName',
+        imageUrl: 'profilePhoto1.png',
+        id: uuidv4(),
+      },
+      createDate: new Date(),
+      updatedDate: new Date(),
+    };
+
+    try {
+      const response = await TicketService.create(transformedData);
+      await dispatch(fetchTicketAsync(apiUrl));
+      return response.data;
+    } catch (error) {
+      Notify('Ticket creation error');
+    }
+  },
+);
+export const updateTicketAsync = createAsyncThunk(
+  'tickets/updateTicket',
+  async ({ id, data }: { id: string; data: IUpdateTickets }) => {
+    const transformedData = {
+      ...data,
+      updatedDate: new Date(),
+    };
+
+    try {
+      const response = await TicketService.update(id, transformedData);
+      return response.data;
+    } catch (error) {
+      Notify('Error');
+    }
+  },
+);
+
+export const deleteTicketAsync = createAsyncThunk('tickets/deleteTicket', async (id: string) => {
+  try {
+    const response = await TicketService.delete(id);
+    return response;
```

Comment:
Do we need to refetch data again after removing?

Reply:
No it is rerendered on the page.

Reply:
We have 10 items and then we remove one. There are just 9 left. So that's why I think we need to fetch all data again